tests/pubsub/testgossipsub.nim

```diff
@@ -92,4 +91,146 @@ suite "GossipSub":
     expect(CatchableError):
       await gossipSub.rpcHandler(peer, @[byte 1, 2, 3])
 
-    await gossipSub.switch.stop()
+  asyncTest "Peer is disconnected and rate limit is hit when overhead rate limit is exceeded":
+    # Given a GossipSub instance with one peer
+    const topic = "foobar"
```

Comment:
Consider making this a module-scoped `const`, also fix places where `topic` is a mutable variable (`let topic`) above.